File: apps/web/src/types/bookmark.ts
// Bookmark represents a saved bookmark
export interface Bookmark {
  id: string;
  title: string;
  url: string;
  created_at: string;
}

// Request body for creating a bookmark
export interface CreateBookmarkRequest {
  title: string;
  url: string;
}

// Request body for updating a bookmark
export interface UpdateBookmarkRequest {
  title?: string;
  url?: string;
}

File: apps/backend/internal/model/bookmark.go
package model

import "time"

// Bookmark represents a saved bookmark
type Bookmark struct {
	ID        string    `json:"id"`
	Title     string    `json:"title"`
	URL       string    `json:"url"`
	CreatedAt time.Time `json:"created_at"`
}

// CreateBookmarkRequest represents the request body for creating a bookmark
type CreateBookmarkRequest struct {
	Title string `json:"title" binding:"required"`
	URL   string `json:"url" binding:"required"`
}

// UpdateBookmarkRequest represents the request body for updating a bookmark
type UpdateBookmarkRequest struct {
	Title string `json:"title"`
	URL   string `json:"url"`
}
